lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php

```diff
@@ -1398,6 +1404,99 @@ public function getSqlResultSetMappings()
         return $this->sqlResultSetMappings;
     }
 
+    /**
+     * Checks whether given property has type
+     *
+     * @param string $name Property name
+     *
+     * @return bool
+     */
+    private function isTypedProperty($name): bool
+    {
+        return PHP_VERSION_ID >= 70400
+               && isset($this->reflClass)
+               && $this->reflClass->hasProperty($name)
+               && $this->reflClass->getProperty($name)->hasType();
+    }
+
+    /**
+     * Validates & completes the given field mapping based on typed property.
+     *
+     * @param  array  $mapping The field mapping to validate & complete.
+     *
+     * @return void
+     */
+    private function _validateAndCompleteTypedFieldMapping(array &$mapping)
```

Comment:
don't pass by reference please but return the array again. This is how other `_validate` methods in this class do it already, so that stay consistent.

Reply:
OK, I have changed _validateAndCompleteFieldMapping too while at it - it was still using reference.